src/librbd/TaskFinisher.h

```diff
@@ -45,6 +45,7 @@ class TaskFinisher {
     typename TaskContexts::iterator it = m_task_contexts.find(task);
     if (it != m_task_contexts.end()) {
       delete it->second.first;
+      m_safe_timer->cancel_event(it->second.second);
```

Comment:
complete() does find later tasks, though, because all Task objects with the same callback compare equal.